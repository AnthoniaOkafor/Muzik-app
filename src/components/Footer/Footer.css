footer{
    background: #1ABC9C 0% 0% no-repeat padding-box;
    border-radius: 10px;
    opacity: 1;
    margin: 0% 5% 12% 5% ;
    /* change padding to affect how social icons look */
    padding: 3% 2% 5% 4%;
    color: white;
    font-size: 125%;

}li{
    list-style-type: none;
}
.footerLink {
    display: flex;
}
/* This section is for different plaform Icon */
.download{
    font-size: 18px;
    margin: -110px 0px 0px 64%;
    
}
.download > ul > li{
    display: inline;
}
.download > ul {
    margin-top:17px;
}
.download > .listOfLink > li > a > img{
    width: 30px;
    padding-left: 18px;
}
/* This section is for different social media Icon */
.footerLink > span > ul {
    background-color: white;
    position: absolute;
    padding: 10px 15px;
    top: 9%;
    left: 97.1%;
}
.footerLink > span > .listOfLink > li > a > img{
    width: 30px;
    padding: 18px 0px;
    display: block
}
.footerLink {
    position: relative;

}
/* This section is for List of Links e.g blog */
.linkGroup > ul > li > a{
    text-decoration: none;
    color: white;
    line-height: 36px;
    font-size: 18px;
}
/* inserts space before every list */
.linkGroup > .listOfLink{
    margin: 20px 0px 20px;
}
/* inserts space after every list */
.linkGroup {
    margin-right: 21%;
}



