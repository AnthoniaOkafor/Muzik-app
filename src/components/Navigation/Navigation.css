.header {
    display: flex;
    justify-content: flex-end;
    padding: 5px 5% 55px;
    margin-bottom: 0px;
    
}

.navList {
padding: 0px 10px;
font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
color: #7f8c8d !important;
font-size: 28px;
}

.navList > a {
    text-decoration: none;
}

.navList > a:hover {
    color: #1ABC9C
}

