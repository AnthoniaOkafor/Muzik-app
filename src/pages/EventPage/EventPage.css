body {
    overflow-x: hidden;
}

.songDetails {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}

.photoCover {
    flex-basis: 60%;
}

.eventDetail {
    flex-basis: 35%;
}

@media (max-width: 740px) {
    .eventDetail {
      order: 1;
      flex-basis: 100%;
      margin-left: 45px;
      margin-top: 15px;
    }
    .photoCover {
      order: 2;
      flex-basis: 100%;
      margin-right: 45px;
    }
  }