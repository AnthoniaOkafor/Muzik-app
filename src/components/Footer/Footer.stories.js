import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs/register';
import Footer from '.';


storiesOf('Footer', module)
	.addDecorator(
		withInfo({
			text: "This is a footer component"
		})
	)
	.add('footer', () => {
        return <Footer />
    })