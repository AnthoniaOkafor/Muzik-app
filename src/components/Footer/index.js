import React from "react"
import Icon from "../Icon/";
import ListOfLink from "../ListOfLink";

import Header from "../Header";
import Text from "../Text"
import "./Footer.css"

const android = "./assets/android.svg"
const apple = "./assets/apple.svg"
export const facebook = "./assets/facebook.svg"
const linkedin = "./assets/linkedin.svg"
const youtube = "./assets/youtube.svg"

export const platform = [{
        item: <Icon src={apple} alt={'apple icon'}/>,
        link: "/lfdjd"
    },{
        item: <Icon src={android} alt={'android icon'}/>,
        link: "#"
    }]
export let socials = [{
        item: <Icon src={facebook} alt={'facebook icon'}/>,
        link: 'https://www.facebook.com'
    },{
        item: <Icon src={linkedin} alt={'linkedin icon'}/>,
        link: 'https://www.linkedin.com'
    },{
        item: <Icon src={youtube} alt={'youtube icon'}/>,
        link: 'https://www.youtube.com'
    }]
export let link =[{
        Categories: ["All events", "Weddings", "Birthdays", "Karoake", "Clubs", "Get togethers"],
        Resources: ["Media resources", "Terms & conditions", "Privacy policy"],
        QuickLinks: ["Blogs", "Company", "Pricing"],
    }]

const Footer = () => { 

    let group = Object.entries(link[0]).map(([key, value]) => {
        return (
            <div key={key} className="linkGroup">
                <Header heading={key} size={3} color={"white"}/>
                {<ListOfLink item={value}/>}
            </div>
        )
    })
    
    return (
        <footer>
            <div className="footerLink">
             {group}
             <span>
             <ListOfLink item={socials}/>
             </span>
             
            </div>
            <div className="download">
                <Text text="Download App" size="extrasmall"/>
                <ListOfLink item={platform}/>
            </div>
        </footer>
    )

}

export default Footer