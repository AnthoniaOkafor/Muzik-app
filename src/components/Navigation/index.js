import React from "react";
import './Navigation.css';



const Navigation = ({navname}) => {
    //const navItems = [{name:"Home"}, {name:"Organise"}, {name:"How it works"}, {name:"Sign in"}]
    const navItems = [
        {name:"Home"}, 
        {name:"Organise", subMenus: ['Submenu 2-1', 'Submenu 2-2', 'Submenu 2-3' ]},
        {name:"How it works", subMenus: ['Submenu 3-1', 'Submenu 3-2', 'Submenu 3-3' ]},
        {name:"Sign in", subMenus: ['Submenu 4-1', 'Submenu 4-2', 'Submenu 4-3' ]}
        ]

    
    return (
        <header className="header">
             {navItems.map(menuItem => {
                 if (menuItem.subMenus !== undefined) {
                    return (
                        <ul className="menuicon" key = {menuItem.id}><i className="fas fa-caret-down" >{menuItem.name}</i>
                        {menuItem.subMenus.map(subMenu => {
                            return (<li className="submenu" key={subMenu.id}>{subMenu}</li>)
                            
                        })}
                        </ul>
                    );
             } else {
                 return <ul key={menuItem.id}>{menuItem.name}</ul>;
             }
            })}

        </header>
    );
}



export default Navigation
