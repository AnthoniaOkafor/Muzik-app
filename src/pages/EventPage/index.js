import React from "react";
import Footer from "../../components/Footer";
import Navigation from "../../components/Navigation";
import EventDetail from "../../components/EventDetail";


export const EventPage = () => {
  return (
    <div className="event-page">
      <Navigation />
      <EventDetail 
      about={"DJ Spinal shutting Down popular club in Yaba"}
      time={"8:00 PM - Dawn"}
      date={"28th January, 2020"}/>
      <Footer />
      
    </div>
  );
};

export default EventPage;
