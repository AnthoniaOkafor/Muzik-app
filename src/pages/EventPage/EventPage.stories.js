import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import EventPage from "./index";

storiesOf("EventPage", module)
  .addDecorator(
    withInfo({
      text: "This is the app's event page"
    })
  )
  .addDecorator(story => {
    return <div>{story()}</div>;
  })
  .add("EventPage", () => (
    <Router>
      <EventPage />
    </Router>
  ));
