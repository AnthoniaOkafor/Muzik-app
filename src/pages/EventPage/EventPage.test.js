import React from "react";
import { shallow } from "enzyme";
import EventPage from "./index";
import Footer from "../../components/Footer";
import Navigation from "../../components/Navigation";

describe("Event Page render", () => {
  const wrapper = shallow(<EventPage />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the Navigation Component", () => {
    const NavigationComponent = wrapper.find(Navigation);
    expect(NavigationComponent).toHaveLength(1);
  });

  it("renders the Footer Component", () => {
    const FooterComponent = wrapper.find(Footer);
    expect(FooterComponent).toHaveLength(1);
  });
});
