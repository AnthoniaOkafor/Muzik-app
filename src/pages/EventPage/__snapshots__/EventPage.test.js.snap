// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Page render renders correctly 1`] = `ShallowWrapper {}`;
